CRM/Utils/File.php

```diff
@@ -769,13 +776,21 @@ public static function findFiles($dir, $pattern, $relative = FALSE) {
           }
         }
       }
+      // Find subdirs to recurse into.
       if ($dh = opendir($subdir)) {
         while (FALSE !== ($entry = readdir($dh))) {
           $path = $subdir . DIRECTORY_SEPARATOR . $entry;
-          if ($entry{0} == '.') {
-            // ignore
-          }
-          elseif (is_dir($path)) {
+          // Exclude . (self) and .. (parent) to avoid infinite loop.
+          // Exclude configured exclude dirs.
+          // Exclude dirs we can't read.
+          // Exclude anything that's not a dir.
+          if (
+            $entry !== '.'
+            && $entry !== '..'
+            && ($excludeDirsPattern && !preg_match($excludeDirsPattern, $path))
```

Comment:
I like the idea (and I think this will still work on windows even).  Just for this line is the purpose of checking $excludeDirsPattern so that someone could set it to `''` or `NULL` and then it wouldn't skip any folders? Since then I think it should be `&& (empty($excludeDirsPattern) || !preg_match($excludeDirsPattern, $path))`